subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/DetachedConfigurationsIntegrationTest.groovy

```diff
@@ -65,4 +66,99 @@ class DetachedConfigurationsIntegrationTest extends AbstractIntegrationSpec {
         expect:
         run "checkDependencies"
     }
+
+    def "detached configurations may have project dependencies"() {
+        given:
+        settingsFile << "include 'other'"
+        buildFile << """
+            plugins {
+                id 'java-library'
+            }
+            
+            def detached = project.configurations.detachedConfiguration()
+            detached.dependencies.add(project.dependencies.create(project(':other')))
+           
+            task checkDependencies {
+                doLast {
+                    assert detached.resolvedConfiguration.getFirstLevelModuleDependencies().moduleName.contains('other')
+                    assert detached.resolvedConfiguration.resolvedArtifacts.collect { it.file.name }.contains("other.jar")
+                }
+            }
+        """
+
+        file("other/build.gradle") << """
+            plugins {
+                id 'java-library'
+            }
+        """
+
+        expect:
+        run "checkDependencies"
+    }
+
+    // This behavior will be removed in Gradle 9.0
+    @Deprecated
+    def "detached configurations may have project dependency on self"() {
```

Comment:
Is this really testing anything different than the one below it? This seems to just test a subset of whats below. I think this test is confusing cause it doesn't make it clear that "a self-project dependency on a detached configuration resolves to that same detached configuration". 

Someone who doesnt get that fact (which is captured in the test below) may be led to believe that we don't want detached configurations to depend on their own project at all. 

In reality, the behavior of a detached configuration depending on `project` is completely wrong here. It should not resolve to itself, but rather resolve to the "real" component with the java output, just like it does with `project(':other')`. 

In terms of actionable changes here, I just think this test doesn't add extra value but the one below is definitely good to verify existing (but not desired) behavior. 

It might be worth writing this up as an issue

Reply:
Agreed, I'll remove this second test.  Issue here: https://github.com/gradle/gradle/issues/24126